### PR TITLE
Fix fusion order bug

### PR DESCRIPTION
**Pull request by zrphercule**
Same problem with: https://github.com/pytorch/tvm/pull/88
When fusing, after a node is changed, we need to recursively re-merge all its neighbor nodes, instead of only searching the afterward nodes.
Here since we dont care about the time complexity, we just simply search all operators again (and again and again... O(N^2) worst case but still tolerable.)

### CODE REVIEW

File: torch_glow/src/GlowFuser.cpp [hunk 1]
```diff
@@ -185,26 +185,33 @@ c10::optional<torch::jit::Node *> tryMerge(torch::jit::Node *consumer,
...
 void GlowCustomFuse(std::shared_ptr<torch::jit::Graph> graph, isSupportFunc fn,
                     at::Symbol kind) {
   torch::jit::AliasDb aliasDb(graph);
   auto block = graph->block();
 
-  for (auto it = block->nodes().rbegin(); it != block->nodes().rend();) {
-    it = getNewNode(*it, aliasDb, block, fn, kind);
-  }
+  bool is_changed;
+  do {
```
Comment (jackm321) [inline torch_glow/src/GlowFuser.cpp]: why not just use a while loop like in https://github.com/pytorch/tvm/pull/88/files?

**Comment by yinghai** [comment]
Yeah, I've brought up this several times. We should just have once piece of code.

**Comment by jackm321** [comment]
@yinghai or @bwasti do you have any suggestions for this? Is there some place in pytorch we can put these things?

File: torch_glow/src/GlowFuser.cpp (hunk 1 above)
Comment (zrphercule) [reply torch_glow/src/GlowFuser.cpp]: o it is same thing...

**Comment by zrphercule** [comment]
> @yinghai or @bwasti do you have any suggestions for this? Is there some place in pytorch we can put these things?

Think we can have this for now, and migrate it to pytorch at some time, or when we need to do some significant change

**Pull request closed without merging**